src/resolvers/index.js

```diff
@@ -1,25 +1,24 @@
 'use strict'
 
-const Multiaddr = require('..') // eslint-disable-line  no-unused-vars
 const protocols = require('../protocols-table')
 
 const { code: dnsaddrCode } = protocols('dnsaddr')
 
+// TODO `addr` type needs https://github.com/microsoft/TypeScript/issues/41672
 /**
  * Resolver for dnsaddr addresses.
  *
- * @param {Multiaddr} addr
+ * @param {any} addr
```

Comment:
Why not a multiaddr?

Reply:
theres a TODO above explaining we can't get the Multiaddr type because of this issue https://github.com/microsoft/TypeScript/issues/41672

Reply:
Ah right! This is not nice as we will need to update this again after, I might wait for this to be solved before getting libp2p PR updating this merged in.
But we can move forward with this module here.

Reply:
the only way to solve this now, is to do a bigger breaking change and default export the Multiaddr class directly. this would remove the ability to `multiaddr('/ip..')` and for `new Multiaddr('')`

should we do this ?

Reply:
i can add a static method to the Multiaddr class called `multiaddr` and users would only need to change from `const multiaddr = require('multiaddr)` to `const { multiaddr } = require('multiaddr)`

Reply:
I would like to have the constructor exported by default to be honest. But, I would wait on @jacobheun input before moving that way as this has a impact on a large number of modules.

Reply:
yeah having the constructor exported directly makes a bunch of things easier and documentation looks a lot better.

Reply:
I'm in favor of exporting the constructor, we should make sure we have clear migration docs for people. We should also add a static method for creation, this would make it easy to hot swap the two and allow us to be more picky about the constructor args. Since this is in the multiformats org it might be good to be consistent with the cid update effort, https://github.com/multiformats/js-multiformats/pull/23, and call it `asMultiaddr`